### ISSUE TITLE
Error Linux Instalations

### ISSUE THREAD

**Issue by dpadron2** [cartologic/cartoview#4]
Hello

I need help to solve the following error:

Installed /usr/local/lib/python2.7/dist-packages/OWSLib-0.10.0-py2.7.egg
Error: Installed distribution OWSLib 0.10.0 conflicts with requirement OWSLib <0.9.0

I'm using Ubuntu 14.04 - Geonode 2.4

I appreciate the help

Thank you

**Comment by kamalseisy** [comment]
Hello
 just clone cartoview and edit the local_settings.py then run using the same python environment that runs geonode

**Comment by dpadron2** [comment]
Hi Kamalseisy

The following error is still generated:

I am doing the following:

1.- apt-get install geonode

2.- I am located in the /usr/local/lib/python2.7/dist-packages/geonode/cartoview directory

3.- In directory /cartoview/ execute: mv local_setting_sample.py local_setting.py

4.- I execute python setup.py install

5.- Installed /usr/local/lib/python2.7/dist-packages/cartoview-0.2-py2.7.egg
Processing dependencies for cartoview == 0.2
Error: Installed distribution OWSLib 0.10.0 conflicts with requirement OWSLib <0.9.0

It may indicate that I am doing wrong.

I appreciate your help

**Comment by kamalseisy** [comment]
Hi Diego,

You don't need to run setup.py as you installed geonode. just run the following
cd geonode
paver start_geosever
cd cartoview
python manage.py runserver

Regards

**Comment by dpadron2** [comment]
Friend Kamal I have not been able to install Cartoview in geonode2.4 -
ubuntu 14.04 Can you help me with a more explicit guide to the
installation? I thank you for all the help, I am very interested in using
Cartoview for teaching purposes at the University of Alcalá Madrid Spain.
Thanks for your help.

*Diego J. Padrón*
*Universidad de Alcalá*
*Twitter: *@diegojpadron
*http://giscode.net <http://giscode.net>*


2016-11-27 2:59 GMT-04:00 Kamal Alseisy <notifications@github.com>:

**Comment by ahmednoureldeen** [comment]
Hi Diego,

to install Geonode on Ubunu 14.04  just follow  [this instructions](http://docs.geonode.org/en/master/tutorials/install_and_admin/geonode_install/install_geonode_application.html)
kindly verify that Geonode installation is working by the following commands:
```
cd <geonode_folder>
python manage.py validate
```

To install Cartoview follow these commands:
`git clone https://github.com/cartologic/cartoview.git`

> there is no specific path to clone Cartoview 

```
sudo pip install south
sudo pip install django-cors-header==1.1.0
sudo pip install django-bootstrap3==6.2.2

cd cartoview 
python manage.py syncdb --all
python manage.py runserver

```

I hope this helps,

**Comment by dpadron2** [comment]
Hello Ahmednoureldeen

I have followed your instructions, but I have remained here:

root@IDE:/home/geonode/geonode/cartoview# python manage.py check
Traceback (most recent call last):
  File "manage.py", line 11, in <module>
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/__init__.py", line 367, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/__init__.py", line 341, in execute
    django.setup()
  File "/usr/local/lib/python2.7/dist-packages/django/__init__.py", line 27, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/usr/local/lib/python2.7/dist-packages/django/apps/registry.py", line 85, in populate
    app_config = AppConfig.create(entry)
  File "/usr/local/lib/python2.7/dist-packages/django/apps/config.py", line 90, in create
    module = import_module(entry)
  File "/usr/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
  File "/usr/local/lib/python2.7/dist-packages/polymorphic/__init__.py", line 11, in <module>
    from .polymorphic_model import PolymorphicModel
  File "/usr/local/lib/python2.7/dist-packages/polymorphic/polymorphic_model.py", line 19, in <module>
    from django.contrib.contenttypes.models import ContentType
  File "/usr/local/lib/python2.7/dist-packages/django/contrib/contenttypes/models.py", line 138, in <module>
    class ContentType(models.Model):
  File "/usr/local/lib/python2.7/dist-packages/django/db/models/base.py", line 105, in __new__
    app_config = apps.get_containing_app_config(module)
  File "/usr/local/lib/python2.7/dist-packages/django/apps/registry.py", line 237, in get_containing_app_config
    self.check_apps_ready()
  File "/usr/local/lib/python2.7/dist-packages/django/apps/registry.py", line 124, in check_apps_ready
    raise AppRegistryNotReady("Apps aren't loaded yet.")
django.core.exceptions.AppRegistryNotReady: Apps aren't loaded yet.

**Comment by dpadron2** [comment]
Hi

A new error has been generated

I'm using django 1.8.2

root@IDE:/home/geonode/geonode/cartoview# python manage.py syncdb --all
Traceback (most recent call last):
  File "manage.py", line 11, in <module>
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/__init__.py", line 338, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/__init__.py", line 312, in execute
    django.setup()
  File "/usr/local/lib/python2.7/dist-packages/django/__init__.py", line 18, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/usr/local/lib/python2.7/dist-packages/django/apps/registry.py", line 85, in populate
    app_config = AppConfig.create(entry)
  File "/usr/local/lib/python2.7/dist-packages/django/apps/config.py", line 86, in create
    module = import_module(entry)
  File "/usr/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
  File "/usr/local/lib/python2.7/dist-packages/polymorphic/__init__.py", line 11, in <module>
    from .polymorphic_model import PolymorphicModel
  File "/usr/local/lib/python2.7/dist-packages/polymorphic/polymorphic_model.py", line 22, in <module>
    from .base import PolymorphicModelBase
  File "/usr/local/lib/python2.7/dist-packages/polymorphic/base.py", line 15, in <module>
    from .manager import PolymorphicManager
  File "/usr/local/lib/python2.7/dist-packages/polymorphic/manager.py", line 8, in <module>
    from polymorphic.query import PolymorphicQuerySet
  File "/usr/local/lib/python2.7/dist-packages/polymorphic/query.py", line 13, in <module>
    from .query_translate import translate_polymorphic_filter_definitions_in_kwargs, translate_polymorphic_filter_definitions_in_args
  File "/usr/local/lib/python2.7/dist-packages/polymorphic/query_translate.py", line 10, in <module>
    from django.db.models.related import RelatedObject
ImportError: No module named related

**Comment by ahmednoureldeen** [comment]
Hi,

Geonode 2.4 is not working on django 1.8.2
It supports django 1.6.11 and so cartoview untill 2 days ago .
now cartoview supports geonode 2.5 which is support django 1.8.2, so make sure to use compatible packages.

**Comment by dpadron2** [comment]
Hello

I followed the instructions and followed these errors:

root@IDE:/home/geonode/geonode/cartoview# python manage.py syncdb --all
Usage: manage.py syncdb [options] 

Create the database tables for all apps in INSTALLED_APPS whose tables haven't already been created.

manage.py: error: no such option: --all

----------------------------------------------

root@IDE:/home/geonode/geonode/cartoview# paver sync
---> pavement.sync
python manage.py syncdb --noinput
Traceback (most recent call last):
  File "manage.py", line 11, in <module>
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/__init__.py", line 399, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/__init__.py", line 392, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/base.py", line 242, in run_from_argv
    self.execute(*args, **options.__dict__)
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/base.py", line 280, in execute
    translation.activate('en-us')
  File "/usr/local/lib/python2.7/dist-packages/django/utils/translation/__init__.py", line 130, in activate
    return _trans.activate(language)
  File "/usr/local/lib/python2.7/dist-packages/django/utils/translation/trans_real.py", line 188, in activate
    _active.value = translation(language)
  File "/usr/local/lib/python2.7/dist-packages/django/utils/translation/trans_real.py", line 177, in translation
    default_translation = _fetch(settings.LANGUAGE_CODE)
  File "/usr/local/lib/python2.7/dist-packages/django/utils/translation/trans_real.py", line 159, in _fetch
    app = import_module(appname)
  File "/usr/local/lib/python2.7/dist-packages/django/utils/importlib.py", line 40, in import_module
    __import__(name)
ImportError: No module named storages


Captured Task Output:
---------------------

---> pavement.sync
python manage.py syncdb --noinput

Build failed running pavement.sync: Subprocess return code: 1

--------------------------------------------------

Some help? I do not know what to try, I can not get cartoview installed

Install Geonode on Ubuntu 14.04


I await your response and thank you for the attention

**Comment by dpadron2** [comment]
Hello
I have tested with Ubuntu 14.04 + Geonode 2.5. 

Another error is generated:

python manage.py syncdb --noinput
System check identified some issues:

WARNINGS:
app_manager.App.tags: (fields.W340) null has no effect on ManyToManyField.
There is no South database module 'south.db.sqlite3' for your database. Please either choose a supported database, check for SOUTH_DATABASE_ADAPTER[S] settings, or remove South from INSTALLED_APPS.

Some help

Thank you

**Comment by ahmednoureldeen** [comment]
just remove South from the installed apps it is built into the core of Django from 1.7 upwards.

**Comment by dpadron2** [comment]
Hi other error

python manage.py syncdb
System check identified some issues:

WARNINGS:
app_manager.App.tags: (fields.W340) null has no effect on ManyToManyField.
Traceback (most recent call last):
  File "manage.py", line 11, in <module>
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/__init__.py", line 338, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/__init__.py", line 330, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/base.py", line 390, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/base.py", line 441, in execute
    output = self.handle(*args, **options)
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/commands/syncdb.py", line 25, in handle
    call_command("migrate", **options)
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/__init__.py", line 120, in call_command
    return command.execute(*args, **defaults)
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/base.py", line 441, in execute
    output = self.handle(*args, **options)
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/commands/migrate.py", line 93, in handle
    executor = MigrationExecutor(connection, self.migration_progress_callback)
  File "/usr/local/lib/python2.7/dist-packages/django/db/migrations/executor.py", line 19, in __init__
    self.loader = MigrationLoader(self.connection)
  File "/usr/local/lib/python2.7/dist-packages/django/db/migrations/loader.py", line 47, in __init__
    self.build_graph()
  File "/usr/local/lib/python2.7/dist-packages/django/db/migrations/loader.py", line 294, in build_graph
    _reraise_missing_dependency(migration, parent, e)
  File "/usr/local/lib/python2.7/dist-packages/django/db/migrations/loader.py", line 264, in _reraise_missing_dependency
    raise exc
django.db.migrations.graph.NodeNotFoundError: Migration app_manager.0001_initial dependencies reference nonexistent parent node (u'base', u'0007_auto_20160915_0944')

**Comment by dpadron2** [comment]
Other Error in Ubuntu 14.04 + Geonode 2.5 + Django 1.8.1

paver sync
---> pavement.sync
python manage.py syncdb --noinput
Operations to perform:
  Synchronize unmigrated apps: tasks, modeltranslation, django_extensions, djcelery, polymorphic, treebeard, gis, pinax_theme_bootstrap_account, django_forms_bootstrap, catalogue, humanize, geoserver, storages, mptt, friendlytagloader, staticfiles, geoexplorer, sitemaps, proxy, pinax_theme_bootstrap, pagination, autocomplete_light, messages, leaflet, api, social, security
  Apply all migrations: dialogos, people, tastypie, agon_ratings, documents, actstream, sites, maps, contenttypes, announcements, user_messages, layers, sessions, guardian, auth, base, groups, services, account, admin, upload, avatar, taggit
Synchronizing apps without migrations:
  Creating tables...
    Running deferred SQL...
  Installing custom SQL...
Running migrations:
  No migrations to apply.
  Your models have changes that are not yet reflected in a migration, and so won't be applied.
  Run 'manage.py makemigrations' to make new migrations, and then re-run 'manage.py migrate' to apply them.
python manage.py loaddata sample_admin.json
Installed 1 object(s) from 1 fixture(s)

**Comment by ahmednoureldeen** [comment]
your last message has no error??
just runserver..

if you still have unmigrated models .. clear your db and run paver sync only.

**Comment by dpadron2** [comment]
Hello Ahmed,

I appreciate your answers but I do not advance :-(

Unfortunately I continue with the problem.

 sudo paver sync
---> pavement.sync
python manage.py syncdb --noinput
Operations to perform:
  Synchronize unmigrated apps: tasks, modeltranslation, django_extensions, djcelery, polymorphic, treebeard, gis, pinax_theme_bootstrap_account, django_forms_bootstrap, catalogue, humanize, geoserver, storages, mptt, friendlytagloader, staticfiles, geoexplorer, sitemaps, proxy, pinax_theme_bootstrap, pagination, autocomplete_light, messages, leaflet, api, social, security
  Apply all migrations: dialogos, people, tastypie, agon_ratings, documents, actstream, sites, maps, contenttypes, announcements, user_messages, layers, sessions, guardian, auth, base, groups, services, account, admin, upload, avatar, taggit
Synchronizing apps without migrations:
  Creating tables...
    Running deferred SQL...
  Installing custom SQL...
Running migrations:
  No migrations to apply.
  Your models have changes that are not yet reflected in a migration, and so won't be applied.
  Run 'manage.py makemigrations' to make new migrations, and then re-run 'manage.py migrate' to apply them.
python manage.py loaddata sample_admin.json
Installed 1 object(s) from 1 fixture(s)
ideadm@ide:/home/geonode/geonode/cartoview$ 

------------------------------

I have uninstalled django 1.8.1 and reinstalled it, I deleted sudo rm -rf /usr/lib/python2.7/dist-packages/django

I have executed paver sync

But I still can not make progress.

If you want to access the server, I can send you the parameters of access to an email, to get your help.

Thank you
